drv/gimlet-seq-server/src/payload.rs

```diff
@@ -0,0 +1,3812 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+///
+/// Iterate over a configuration payload for a Renesas 8A3XXXX clock
+/// generator.  This code was generated by "humility rencm -g" given
+/// Aardvark output generated by running Renesas configuration software.
+///
+#[rustfmt::skip]
+pub fn idt8a3xxxx_payload<E>(
+    mut func: impl FnMut(&[u8]) -> Result<(), E>
+) -> Result<(), E> {
+
+    // PAGE_ADDR = 0xc000
+    func(&[
```

Comment:
I suggested this before, so maybe you've already tried it and I missed the response, but:

I suspect this code would be significantly smaller if you restructured this as an array of things to send, and then called `func` in a loop. That is,

```rust
const PAYLOAD: &[&[u8]] = &[ &[0xfc ..], ..];
for chunk in PAYLOAD {
    func(chunk); // lol it rhymes
}
```

Would you be willing to try this and check the text size difference?

Reply:
Yes, will do!

Reply:
Good intuition! This is a significant savings. Here is the code before:

```
section              size        addr
.text               27420   134348800
.rodata             33188   134376220
.addr_table             0   134409408
.data                1292   536938496
.bss                    0   536939788
```

And afterwards:

```
section              size        addr
.text               15292   134348800
.rodata             37300   134364092
.addr_table             0   134401392
.data                1292   536938496
.bss                    0   536939788
```

So more `.rodata` -- but much, much less text, and significant net savings:

```
$ size ./gimlet_seq ./gimlet_seq-after 
   text	   data	    bss	    dec	    hex	filename
  60608	   1292	      0	  61900	   f1cc	./gimlet_seq
  52592	   1292	      0	  53884	   d27c	./gimlet_seq-after
```

Clearly a very good change; thank you for the suggestion!